src/main/java/org/zalando/nakadi/service/subscription/state/ClosingState.java

```diff
@@ -39,9 +43,10 @@ public void onExit() {
 
     @Override
     public void onEnter() {
-        final long timeToWaitMillis = getParameters().commitTimeoutMillis - (System.currentTimeMillis()
-                - lastCommitMillis);
-        if (timeToWaitMillis > 0) {
+        final long timeToWaitMillis = getParameters().commitTimeoutMillis -
+                (System.currentTimeMillis() - lastCommitSupplier.getAsLong());
+        uncommittedOffsets = uncommittedOffsetsSupplier.get();
+        if (!uncommittedOffsets.isEmpty() && timeToWaitMillis > 0) {
```

Comment:
Is this new check for `!uncommittedOffsets.isEmpty()` part of the bug fix or an optimization?